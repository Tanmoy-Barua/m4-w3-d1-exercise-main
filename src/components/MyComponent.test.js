import React from 'react';
import { shallow } from 'enzyme';
import MyComponent from './MyComponent';

describe("MyComponent", () => {
  let component;
  let form;

  beforeEach(() => {
    component = shallow(<MyComponent />);
    form = component.find('input');
  });

  it("should render correctly", () => {
    expect(component).toMatchSnapshot();
  });

  it("should render initial layout", () => {
    expect(component.getElements()).toMatchSnapshot();
  });

  it("should create an entry in component state", () => {
    // Simulate input change
    form.simulate('change', { target: { name: 'myName', value: 'myValue' } });

    // Verify the state is defined
    expect(component.state('input')).toBeDefined();
  });

  it("should create an entry in component state with the event value", () => {
    // Simulate input change
    form.simulate('change', { target: { name: 'myName', value: 'myValue' } });

    // Verify the state contains the correct value
    expect(component.state('input')).toEqual('myValue');
  });
});
