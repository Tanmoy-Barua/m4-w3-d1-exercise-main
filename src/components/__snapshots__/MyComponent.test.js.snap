// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render correctly 1`] = `ShallowWrapper {}`;

exports[`MyComponent should render initial layout 1`] = `
[
  <div>
    <input
      onChange={[Function]}
      type="text"
    />
  </div>,
]
`;
